src/main/java/org/openstreetmap/atlas/vectortiles/TippecanoeExporter.java

```diff
@@ -0,0 +1,148 @@
+package org.openstreetmap.atlas.vectortiles;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.PumpStreamHandler;
+import org.apache.commons.exec.util.StringUtils;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.openstreetmap.atlas.geography.atlas.geojson.LineDelimitedGeoJsonConverter;
+import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+import org.openstreetmap.atlas.utilities.time.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This CLI will take a directory full of atlases and export it into an MBTiles file full of Mapbox
+ * vector tiles. It creates intermediary line-delimited GeoJSON, and then it ultimately drives
+ * tippecanoe to do the vector tile creation.
+ *
+ * @author hallahan
+ */
+public final class TippecanoeExporter extends LineDelimitedGeoJsonConverter
+{
+    private static final int EXIT_FAILURE = 1;
+
+    private static final Logger logger = LoggerFactory.getLogger(TippecanoeExporter.class);
+
+    private static final Switch<Path> MBTILES = new Switch<>("mbtiles",
+            "The MBTiles file to which tippecanoe will write vector tiles.", Paths::get,
+            Optionality.REQUIRED);
+
+    private TippecanoeExporter()
+    {
+        this.setJsonMutator(TippecanoeSettings.JSON_MUTATOR);
+    }
+
+    public static void main(final String[] args)
+    {
+        new TippecanoeExporter().run(args);
+    }
+
+    @Override
+    protected int onRun(final CommandMap command)
+    {
+        super.onRun(command);
+
+        if (!hasValidTippecanoe())
+        {
+            logger.error(
+                    "Your system does not have a valid installation of tippecanoe installed in its path.");
+            logger.error("https://github.com/mapbox/tippecanoe");
+
+            System.exit(EXIT_FAILURE);
+        }
+
+        final Path mbtiles = (Path) command.get(MBTILES);
+
+        final Path geojsonDirectory = (Path) command.get(GEOJSON_DIRECTORY);
+        final Path geojson = geojsonDirectory.resolve(TippecanoeSettings.GEOJSON);
+
+        final Boolean overwrite = (Boolean) command.get(OVERWRITE);
+
+        runTippecanoe(geojson, mbtiles, overwrite);
+
+        return 0;
+    }
+
+    @Override
+    protected SwitchList switches()
+    {
+        return super.switches().with(MBTILES);
+    }
+
+    private boolean hasValidTippecanoe()
+    {
+        final CommandLine commandLine = CommandLine.parse("tippecanoe").addArgument("--version");
```

Comment:
Same, use `RunScript` here.

Reply:
Actually even better, the tippecanoe check should live in a shell script in the resource folder: `src/main/resources/path/to/package`

Reply:
@hallahan to solve your problem, try adding the new API we discussed, that calls:
```java
public Process exec(String[] cmdarray, String[] envp, File dir);
```
This will handle tokenizing correctly. @matthieun I think this is a good way to use `RunScript` while still allowing @hallahan to get his code working.

Reply:
Ok, I'll spend more time trying to make this work.

Reply:
Ok, I am now using only RunScript, no more apache commons exec.